src/wg/format/png/types.d

```diff
@@ -0,0 +1,186 @@
+module wg.format.png.types;
+
+package enum ubyte[8] PNG_SIGNATURE = [137, 80, 78, 71, 13, 10, 26, 10];
+
+package enum crcSize = 4;
+
+// All the following structs must not have padding
+align(1):
+
+/**
+ * Each Chunk of data starts with this header and is then followed by length
+ * bytes of data and then 4 more bytes of CRC checksum for data validation.
+ */
+package struct PngChunkHeader
+{
+    uint length;
+    immutable(char)[4] type;
+}
+
+/// Stores a data from one png chunk and a link to the next one
+struct PngChunk {
+    ///
+    char[4] type;
+    ///
+    ubyte[] data;
+    ///
+    PngChunk* next;
+}
+
+package enum char[4] HEADER_CHUNK_TYPE = "IHDR";
+
+/**
+ * Bit depth says how many bits per sample there is. Pixel can have one sample in case of
+ * grayscale images and up to 4 samples in case of RGBA images since each channel is one sample.
+ */
+enum PngBitDepth: ubyte
+{
+    one = 1,
+    two = 2,
+    four = 4,
+    eight = 8,
+    sixteen = 16
+}
+
+// Checks if the given value actually matches one of defined enum values
+package bool isValid(PngBitDepth depth) nothrow @nogc
+{
+    import std.traits: EnumMembers;
+    static foreach (i, member; EnumMembers!PngBitDepth)
+    {
+        if (depth == member) return true;
+    }
+    return false;
+}
+
+/// Defines the options that png supports for representing pixels
+enum PngColorType: ubyte
+{
+    grayscale = 0,
+    rgbColor = 2,
+    paletteColor = 3,
+    grayscaleWithAlpha = 4,
+    rgbaColor = 6
+}
+
+package ubyte[7] pngColorTypeToSamples = [1, 0, 3, 1, 2, 0, 4];
```

Comment:
TLS data?

Reply:
What does that mean :)? Are you saying I should just make it an enum?

Reply:
It's mutable global data, which means it's thread-local allocated.
You should probably use `immutable`, or make it `enum`.